pkg/engine/source/filesystem.go

```diff
@@ -258,18 +258,31 @@ func checkQueryFeatureFlagDisabled(ctx context.Context, metadata map[string]inte
 
 	logger := logger.FromContext(ctx)
 	// Check if the rule is disabled via feature flag
-	disabled, err := queryParameters.FlagEvaluator.EvaluateWithOrgAndCustomVariables("k9-iac-disable-kics-rule", customVariables)
+	ruleIdDisabled, err := queryParameters.FlagEvaluator.EvaluateWithOrgAndCustomVariables("k9-iac-disable-kics-rule", customVariables)
 	if err != nil {
 		// If feature flag evaluation fails, log and continue (fail open)
-		logger.Warn().Err(err).Str("kics_id", kicsIDStr).Msg("Failed to evaluate feature flag for KICS rule")
+		logger.Warn().Err(err).Str("kics_id", kicsIDStr).Str("reason", "id").Msg("Failed to evaluate feature flag for KICS rule")
+	}
+
+	if ruleIdDisabled {
+		logger.Info().Str("kics_id", kicsIDStr).Str("reason", "id").Msg("KICS rule disabled by feature flag")
+		return true
+	}
+
+	// Check if the rule is disabled via feature flag
+	rulePlatformDisabled, err := queryParameters.FlagEvaluator.EvaluateWithOrgAndCustomVariables("k9-iac-disable-kics-platform", customVariables)
+	if err != nil {
+		// If feature flag evaluation fails, log and continue (fail open)
+		logger.Warn().Err(err).Str("kics_id", kicsIDStr).Str("reason", "platform").Msg("Failed to evaluate feature flag for KICS rule")
```

Comment:
Same comment, we should log:
```suggestion
		logger.Warn().Err(err).Str("kics_id", kicsIDStr).Str("feature_flag", featureflags. IacDisableKicsPlatform).Msg("Failed to evaluate feature flag for KICS rule")
```

Also don’t forget to add `IacDisableKicsPlatform` to pkg/featureflags/evaluator.go and pkg/featureflags/local_evaluator.go